stdlib/public/core/StringStorage.swift

```diff
@@ -691,7 +725,10 @@ final internal class __SharedStringStorage
   }
 
   @inline(__always)
-  final internal var isASCII: Bool { return _countAndFlags.isASCII }
+  final internal var isASCII: Bool {
+    get { return _countAndFlags.isASCII }
+    set { _countAndFlags = CountAndFlags(mortalCount: count, isASCII: newValue)}
+  }
```

Comment:
I think it's better to keep this computed get only, and have a `forceSetIsASCII()` or some such, as setting is a high-specialized use case. That method would then perform the invariant check afterwards so we can assert.

Reply:
ok